### PR TITLE
[AutoScheduler] Enable schedule sharing in dispatch context

### PR DESCRIPTION
**Pull request by comaniac**
This is the follow up PR for #7317 to enable the schedule sharing in the auto_scheduler diaptch context.

cc @merrymercy @jcf94

### CODE REVIEW

**Comment by masahi** [comment]
@comaniac Does this mean tuning support for dynamic workload (dynamic batch size etc) is coming soon? I'm very excited for this, that would tremendously help my MaskRCNN!!

**Comment by comaniac** [comment]
> @comaniac Does this mean tuning support for dynamic workload (dynamic batch size etc) is coming soon? I'm very excited for this, that would tremendously help my MaskRCNN!!

Ah this is not the perfect solution for dynamic shape. This is more like a solution to make tuned logs more useful. For example, you can apply the tuning log with batch 1 to all batch sizes. You can even tune several batch sizes in prime numbers to achieve better performance to their multiples. Meanwhile, we do work on the dynamic shape support in auto_scheduler, but it may not be ready to be upstreamed before this summer or fall.

**Comment by jcf94** [comment]
> > @comaniac Does this mean tuning support for dynamic workload (dynamic batch size etc) is coming soon? I'm very excited for this, that would tremendously help my MaskRCNN!!
> 
> Ah this is not the perfect solution for dynamic shape. This is more like a solution to make tuned logs more useful. For example, you can apply the tuning log with batch 1 to all batch sizes. You can even tune several batch sizes in prime numbers to achieve better performance to their multiples. Meanwhile, we do work on the dynamic shape support in auto_scheduler, but it may not be ready to be upstreamed before this summer or fall.

😃  Looking forward to the dynamic shape support, too! It will be great useful.

**Comment by masahi** [comment]
Yes, I believe dynamic tuning and codegen is one of the biggest challenges of TVM this year. I'm glad at least there are folks looking at the problem. MaskRCNN should serve as a good benchmark, it has both dynamic dense (very large) and dynamic conv2d + conv2d transpose.

**Review by merrymercy — APPROVED** [review]

File: python/tvm/auto_scheduler/dispatcher.py [hunk 1]
```diff
@@ -126,18 +127,53 @@ class ApplyHistoryBest(DispatchContext):
...
+            The first key to the best_records.
+        workload_key: str
+            The workload key that can be decoded to workload hash and args.
+
+        Returns
+        -------
+        entry: Dict[str, Any]
+            The entry in best_records with target key and workload hash.
+        workload_hash: str
+            The workload hash.
```
Comment (comaniac) [inline python/tvm/auto_scheduler/dispatcher.py:166]:
```suggestion
            The workload hash decoded from workload_key
```

**Comment by comaniac** [comment]
Thanks @merrymercy

**Pull request merged**